Step2/Install_Win_Features.ps1

```diff
@@ -0,0 +1,86 @@
+# Import the configuration
+Import-Module -Name .\Config.ps1
+
+# Function to handle logging and console output
+function Log-Message {
+    param (
+        [string]$Message
+    )
+    
+    $Timestamp = (Get-Date).ToString("yyyy-MM-dd HH:mm:ss")
+    "$Timestamp - $Message" | Out-File -FilePath $Global:LogFilePath -Append
+    Write-Host $Message
+}
+
+function Install_Win_Features {
+    param (
+        [string[]]$Features
+    )
+    
+    $InstallationSuccessful = $true
+
+    foreach ($Feature in $Features) {
+        try {
+            Log-Message "Installing feature: $Feature"
+            Install-WindowsFeature -Name $Feature -Verbose
+            Log-Message "Feature $Feature installed successfully."
+        } catch {
+            Log-Message "Failed to install feature $Feature. Error: $_"
+            $InstallationSuccessful = $false
+        }
+    }
+
+    return $InstallationSuccessful
+}
+
+function Rename_Server {
+    param (
+        [string]$NewName
+    )
+    
+    try {
+        Log-Message "Renaming server to: $NewName"
+        Rename-Computer -NewName $NewName -Force

```

Comment:
Renmae-Computer comes with a built-in "restart" functionality

Reply:
Handling **Rename-Computer** and **Restart-Computer** as separate steps with individual try-catch blocks might be a better approach. It allows us precise error handling and logging for each operation.
If renaming the server fails, we can address that issue specifically without affecting the restart operation.
Furthermore, it gives us greater control over when the restart occurs.